src/Mvc/Mvc.Abstractions/src/Abstractions/ActionDescriptor.cs

```diff
@@ -38,42 +38,42 @@ public ActionDescriptor()
         /// <summary>
         /// Gets or sets the <see cref="Routing.AttributeRouteInfo"/>.
         /// </summary>
-        public AttributeRouteInfo AttributeRouteInfo { get; set; }
+        public AttributeRouteInfo? AttributeRouteInfo { get; set; }
 
         /// <summary>
         /// The set of constraints for this action. Must all be satisfied for the action to be selected.
         /// </summary>
-        public IList<IActionConstraintMetadata> ActionConstraints { get; set; }
+        public IList<IActionConstraintMetadata>? ActionConstraints { get; set; }
 
         /// <summary>
         /// Gets or sets the endpoint metadata for this action.
         /// This API is meant for infrastructure and should not be used by application code.
         /// </summary>
-        public IList<object> EndpointMetadata { get; set; }
+        public IList<object>? EndpointMetadata { get; set; }
 
         /// <summary>
         /// The set of parameters associated with this action.
         /// </summary>
-        public IList<ParameterDescriptor> Parameters { get; set; }
+        public IList<ParameterDescriptor> Parameters { get; set; } = Array.Empty<ParameterDescriptor>();
```

Comment:
These are not read-only, if we do this we can cause issues to existing code when they call `Add`

Reply:
These get initialized later on by the framework. It's not an issue in actual usage.

Reply:
Sure, I don't really understand why we prefer `Array.Empty<T>` to default! but I don't feel super strongly about it.